wordpress-mobile/WordPress-Android
# Dashboard: unified menu Refactoring

Parent #19270 

- Add ListItemAction to ListItem
- Add logic to hide/show quick link preferences
- Update ShortCutsState to include ListItemAction
- Handle action button click within the Personalize Shortcuts View

Note: RTL will be handled in a separate PR

To test:
- Install the app
- Navigate to jetpack app
- Navigate to dashboard
- Tap on the personalize your home tab card
- Navigate to the shortcuts tab
- Tap on an "-" icon for an item within the Active Shortcuts list 
- ✅ Verify the item moved to the Inactive Shortcuts list
- Tap on an "+" icon for an item within the Inactive Shortcuts list 
- ✅ Verify the item moved to the Active Shortcuts list


## Regression Notes
1. Potential unintended areas of impact
The item doesn't move between the active/inactive lists

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

UI Changes testing checklist:
